src/app/services/search.py

```diff
@@ -189,27 +155,73 @@ def embed_query(self, search_input: str, curr_model: str) -> np.ndarray:
         return cast(np.ndarray, embeddings)
 
     def build_filters(
-        self, filters: Optional[List[int]] = None
+        self, filters: Optional[SearchFilters] = None
     ) -> Optional[qdrant_models.Filter]:
         if filters is None:
             return None
 
-        qdrant_filter: List[qdrant_models.Condition] = [
-            qdrant_models.FieldCondition(
-                key="document_sdg", match=qdrant_models.MatchValue(value=filt)
+        qdrant_filter = []
+        for key, values in filters.model_dump().items():
+            if not values:
+                continue
+
+            qdrant_filter.append(
+                qdrant_models.FieldCondition(
+                    key=key,
+                    match=qdrant_models.MatchAny(any=values),
+                )
             )
-            for filt in filters
-        ]
-        return qdrant_models.Filter(should=qdrant_filter)
+
+        logger.debug("build_filters=%s", qdrant_filter)
```

Comment:
```suggestion
        logger.debug(f"build_filters={qdrant_filter}")
```
I read somewhere the old way with '%s' must be abandoned for format. Idk if it's true but when log is print it's work with format and sometimes not with %s